### PR TITLE
Diz39o

### PR DESCRIPTION
**Pull request by s1mark**

### CODE REVIEW

File: Tolvaj Balazs - DIZ39O/module/vm/provider.tf [hunk 1]
```diff
@@ -0,0 +1,11 @@
+provider "google" {
+    project = var.project_name
+    region  = var.provider_region
+}
+
+terraform {
```
Comment (s1mark) [inline Tolvaj Balazs - DIZ39O/module/vm/provider.tf:6]: unnecessary part

File: Tolvaj Balazs - DIZ39O/beadando/provider.tf [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  region  = var.provider_region
+}
+
+# resource "google_storage_bucket" "bucket" {
+#    name = var.project_neptun_code
+#    location = var.provider_location
+#    uniform_bucket_level_access = true   
+#}
+
+resource "google_compute_disk" "default" {
```
Comment (s1mark) [inline Tolvaj Balazs - DIZ39O/beadando/provider.tf:12]: the two resource would look better in the main.tf not in the provider.tf

File: Tolvaj Balazs - DIZ39O/beadando/outputs.tf [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+output "instance_id" {
+  value = google_compute_instance.default.instance_id
+}
+
+output "instance_name" {
+  value = google_compute_instance.default.name
+}
+
+output "external_ip" {
```
Comment (s1mark) [inline Tolvaj Balazs - DIZ39O/beadando/outputs.tf:17]: nice

**Comment by s1mark** [comment]
The service account and vm creation looks ok. Have you tried running terraform plan?

**Comment by nopizzaa** [comment]
> The service account and vm creation looks ok. Have you tried running terraform plan?

Yes, terraform apply also ran without any problem.

## terraform plan
```console
╭─nopizza@laposdell ~/git/cloud/Tolvaj Balazs - DIZ39O/beadando ‹diz39o●› 
╰─$ terraform plan                                                                                                                               1 ↵

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # google_compute_disk.default will be created
  + resource "google_compute_disk" "default" {
      + creation_timestamp        = (known after apply)
      + id                        = (known after apply)
      + image                     = "debian-11-bullseye-v20220719"
      + label_fingerprint         = (known after apply)
      + last_attach_timestamp     = (known after apply)
      + last_detach_timestamp     = (known after apply)
      + name                      = "diz39o"
      + physical_block_size_bytes = 4096
      + project                   = (known after apply)
      + provisioned_iops          = (known after apply)
      + self_link                 = (known after apply)
      + size                      = 10
      + source_disk_id            = (known after apply)
      + source_image_id           = (known after apply)
      + source_snapshot_id        = (known after apply)
      + type                      = "pd-standard"
      + users                     = (known after apply)
      + zone                      = "europe-central2-a"
    }

  # google_compute_instance.default will be created
  + resource "google_compute_instance" "default" {
      + can_ip_forward          = false
      + cpu_platform            = (known after apply)
      + current_status          = (known after apply)
      + deletion_protection     = false
      + guest_accelerator       = (known after apply)
      + id                      = (known after apply)
      + instance_id             = (known after apply)
      + label_fingerprint       = (known after apply)
      + machine_type            = "f1-micro"
      + metadata_fingerprint    = (known after apply)
      + metadata_startup_script = "echo done >> ~/task"
      + min_cpu_platform        = (known after apply)
      + name                    = "diz39o"
      + project                 = (known after apply)
      + self_link               = (known after apply)
      + tags_fingerprint        = (known after apply)
      + zone                    = "europe-central2-a"

      + boot_disk {
          + auto_delete                = true
          + device_name                = (known after apply)
          + disk_encryption_key_sha256 = (known after apply)
          + kms_key_self_link          = (known after apply)
          + mode                       = "READ_WRITE"
          + source                     = (known after apply)

          + initialize_params {
              + image  = (known after apply)
              + labels = (known after apply)
              + size   = (known after apply)
              + type   = (known after apply)
            }
        }

      + confidential_instance_config {
          + enable_confidential_compute = (known after apply)
        }

      + network_interface {
          + ipv6_access_type   = (known after apply)
          + name               = (known after apply)
          + network            = "default"
          + network_ip         = (known after apply)
          + stack_type         = (known after apply)
          + subnetwork         = (known after apply)
          + subnetwork_project = (known after apply)
        }

      + reservation_affinity {
          + type = (known after apply)

          + specific_reservation {
              + key    = (known after apply)
              + values = (known after apply)
            }
        }

      + scheduling {
          + automatic_restart           = (known after apply)
          + instance_termination_action = (known after apply)
          + min_node_cpus               = (known after apply)
          + on_host_maintenance         = (known after apply)
          + preemptible                 = (known after apply)
          + provisioning_model          = (known after apply)

          + node_affinities {
              + key      = (known after apply)
              + operator = (known after apply)
              + values   = (known after apply)
            }
        }

      + service_account {
          + email  = (known after apply)
          + scopes = [
              + "https://www.googleapis.com/auth/monitoring",
            ]
        }
    }

  # google_service_account.default will be created
  + resource "google_service_account" "default" {
      + account_id   = "diz39o"
      + disabled     = false
      + display_name = "diz39o"
      + email        = (known after apply)
      + id           = (known after apply)
      + member       = (known after apply)
      + name         = (known after apply)
      + project      = (known after apply)
      + unique_id    = (known after apply)
    }

Plan: 3 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + external_ip     = (known after apply)
  + gcloud_cpu_info = (known after apply)
  + instance_id     = (known after apply)
  + instance_name   = "diz39o"
  + project_id      = (known after apply)

────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

Note: You didn't use the -out option to save this plan, so Terraform can't guarantee to take exactly these actions if you run "terraform apply" now.
```

## terraform apply
```console
╭─nopizza@laposdell ~/git/cloud/Tolvaj Balazs - DIZ39O/beadando ‹diz39o●› 
╰─$ terraform apply            

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # google_compute_disk.default will be created
  + resource "google_compute_disk" "default" {
      + creation_timestamp        = (known after apply)
      + id                        = (known after apply)
      + image                     = "debian-11-bullseye-v20220719"
      + label_fingerprint         = (known after apply)
      + last_attach_timestamp     = (known after apply)
      + last_detach_timestamp     = (known after apply)
      + name                      = "diz39o"
      + physical_block_size_bytes = 4096
      + project                   = (known after apply)
      + provisioned_iops          = (known after apply)
      + self_link                 = (known after apply)
      + size                      = 10
      + source_disk_id            = (known after apply)
      + source_image_id           = (known after apply)
      + source_snapshot_id        = (known after apply)
      + type                      = "pd-standard"
      + users                     = (known after apply)
      + zone                      = "europe-central2-a"
    }

  # google_compute_instance.default will be created
  + resource "google_compute_instance" "default" {
      + can_ip_forward          = false
      + cpu_platform            = (known after apply)
      + current_status          = (known after apply)
      + deletion_protection     = false
      + guest_accelerator       = (known after apply)
      + id                      = (known after apply)
      + instance_id             = (known after apply)
      + label_fingerprint       = (known after apply)
      + machine_type            = "f1-micro"
      + metadata_fingerprint    = (known after apply)
      + metadata_startup_script = "echo done >> ~/task"
      + min_cpu_platform        = (known after apply)
      + name                    = "diz39o"
      + project                 = (known after apply)
      + self_link               = (known after apply)
      + tags_fingerprint        = (known after apply)
      + zone                    = "europe-central2-a"

      + boot_disk {
          + auto_delete                = true
          + device_name                = (known after apply)
          + disk_encryption_key_sha256 = (known after apply)
          + kms_key_self_link          = (known after apply)
          + mode                       = "READ_WRITE"
          + source                     = (known after apply)

          + initialize_params {
              + image  = (known after apply)
              + labels = (known after apply)
              + size   = (known after apply)
              + type   = (known after apply)
            }
        }

      + confidential_instance_config {
          + enable_confidential_compute = (known after apply)
        }

      + network_interface {
          + ipv6_access_type   = (known after apply)
          + name               = (known after apply)
          + network            = "default"
          + network_ip         = (known after apply)
          + stack_type         = (known after apply)
          + subnetwork         = (known after apply)
          + subnetwork_project = (known after apply)
        }

      + reservation_affinity {
          + type = (known after apply)

          + specific_reservation {
              + key    = (known after apply)
              + values = (known after apply)
            }
        }

      + scheduling {
          + automatic_restart           = (known after apply)
          + instance_termination_action = (known after apply)
          + min_node_cpus               = (known after apply)
          + on_host_maintenance         = (known after apply)
          + preemptible                 = (known after apply)
          + provisioning_model          = (known after apply)

          + node_affinities {
              + key      = (known after apply)
              + operator = (known after apply)
              + values   = (known after apply)
            }
        }

      + service_account {
          + email  = (known after apply)
          + scopes = [
              + "https://www.googleapis.com/auth/monitoring",
            ]
        }
    }

  # google_service_account.default will be created
  + resource "google_service_account" "default" {
      + account_id   = "diz39o"
      + disabled     = false
      + display_name = "diz39o"
      + email        = (known after apply)
      + id           = (known after apply)
      + member       = (known after apply)
      + name         = (known after apply)
      + project      = (known after apply)
      + unique_id    = (known after apply)
    }

Plan: 3 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + external_ip     = (known after apply)
  + gcloud_cpu_info = (known after apply)
  + instance_id     = (known after apply)
  + instance_name   = "diz39o"
  + project_id      = (known after apply)

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

google_service_account.default: Creating...
google_compute_disk.default: Creating...
google_service_account.default: Creation complete after 2s [id=projects/ekke-cloud/serviceAccounts/diz39o@ekke-cloud.iam.gserviceaccount.com]
google_compute_disk.default: Still creating... [10s elapsed]
google_compute_disk.default: Creation complete after 13s [id=projects/ekke-cloud/zones/europe-central2-a/disks/diz39o]
google_compute_instance.default: Creating...
google_compute_instance.default: Still creating... [10s elapsed]
google_compute_instance.default: Creation complete after 13s [id=projects/ekke-cloud/zones/europe-central2-a/instances/diz39o]

Apply complete! Resources: 3 added, 0 changed, 0 destroyed.

Outputs:

external_ip = "10.186.0.45"
gcloud_cpu_info = "Intel Broadwell"
instance_id = "4555183398960895354"
instance_name = "diz39o"
project_id = "ekke-cloud"
```

**Comment by s1mark** [comment]
Looks good

**Pull request merged**